src/main/java/gov/cabinetoffice/api/mappers/CustomSubmissionMapperImpl.java

```diff
@@ -22,74 +23,80 @@
-				}
-		final String objectKey = grantAttachment.getLocation().split(s3ConfigProperties.getSourceBucket()+"/")[1];
-
-		return s3Service.createPresignedURL(bucketName, objectKey);
-	}
-
-	@Override
-	public LocalDate buildDate(String[] multiResponse) {
-		return SubmissionMapper.super.buildDate(multiResponse);
-	}
-
-	@Override
-	public AddressDTO buildAddress(String[] multiResponse) {
-		return SubmissionMapper.super.buildAddress(multiResponse);
-	}
-
-	@Override
-	public SubmissionDTO submissionToSubmissionDto(Submission submission) {
-		if (submission == null) {
-			return null;
-		}
+    private final GrantAttachmentService grantAttachmentService;
+    private final S3Service s3Service;
+    private final S3ConfigProperties s3ConfigProperties;
+    private final UserService userService;
+
+    @Override
+    public List<SubmissionSectionDTO> mapSections(List<SubmissionSection> sections) {
+        return SubmissionMapper.super.mapSections(sections);
+    }
+
+    @Override
+    public List<SubmissionSectionDTO> submissionSectionListToSubmissionSectionDtoList(
+            List<SubmissionSection> submissionSections) {
+        return SubmissionMapper.super.submissionSectionListToSubmissionSectionDtoList(submissionSections);
+    }
+
+    @Override
+    public List<SubmissionQuestionDTO> submissionQuestionListToSubmissionQuestionDtoList(
+            List<SubmissionQuestion> submissionQuestions) {
+        return SubmissionMapper.super.submissionQuestionListToSubmissionQuestionDtoList(submissionQuestions);
+    }
+
+    @Override
+    public SubmissionQuestionDTO submissionQuestionToSubmissionQuestionDto(SubmissionQuestion submissionQuestion) {
+        return SubmissionMapper.super.submissionQuestionToSubmissionQuestionDto(submissionQuestion);
+    }
+
+    @Override
+    public Object getQuestionResponseByResponseType(SubmissionQuestion submissionQuestion) {
+        return SubmissionMapper.super.getQuestionResponseByResponseType(submissionQuestion);
+    }
+
+    @Override
+    public String buildUploadResponse(SubmissionQuestion submissionQuestion) {
+        final UUID grantAttachmentId = submissionQuestion.getAttachmentId();
+        final GrantAttachment grantAttachment = grantAttachmentService.getGrantAttachmentById(grantAttachmentId);
+        final String bucketName = s3ConfigProperties.getSourceBucket();
+
+        final GrantAttachmentStatus status = grantAttachment.getStatus();
+        return switch (status) {
+            case QUARANTINED -> "This file is not available. Please contact the Grant Applicant.";
+            case AWAITING_SCAN ->
+                    "The URL for this file is not available at the moment. It is undergoing our antivirus process. Please try later.";
+            default -> {
+                final String objectKey = grantAttachment.getLocation().split(bucketName + "/")[1];
+                 yield s3Service.createPresignedURL(s3ConfigProperties.getSourceBucket(), objectKey);
+            }
+        };
```

Comment:
I'd put in an explicit case for AVAILABLE and have the default return a nondescript "This file is not available" message to future proof incase we add additional status to that column